caf-audit/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditTransportClientFactory.java

```diff
@@ -49,49 +50,63 @@ private ElasticAuditTransportClientFactory() {
      * @return TransportClient
      * @throws ConfigurationException exception thrown if host is unknown
      */
-    public static TransportClient getTransportClient(String hostAndPortValues, String clusterName) throws ConfigurationException{
+    public static TransportClient getTransportClient(String hostAndPortValues, String clusterName) throws ConfigurationException {
         final TransportClient transportClient;
 
-        try {
-            Settings settings = Settings.builder()
-                    .put("cluster.name", clusterName).build();
-            transportClient = new PreBuiltTransportClient(settings);
+        if (hostAndPortValues != null && !hostAndPortValues.isEmpty()) {
+            //  Split comma separated list of ES hostname and port values.
+            final String[] hostAndPortArray = hostAndPortValues.split(",");
 
-            if (hostAndPortValues != null && !hostAndPortValues.isEmpty()) {
-                //  Split comma separated list of ES hostname and port values.
-                final String[] hostAndPortArray = hostAndPortValues.split(",");
+            // If there is more than one Elasticsearch host then set the client.transport.sniff to true as we are dealing with a cluster
+            Settings.Builder settingsBuilder = Settings.builder()
+                    .put("cluster.name", clusterName);
+            if (hostAndPortArray.length > 1) {
+                settingsBuilder.put("client.transport.sniff", true);
```

Comment:
Yes the transport client is made aware of nodes, removed or added to the cluster, from the cluster with this option enabled. Our integration tests, that use 1 node ES, started to fail because of an exception indicating that the transport client could not connect to the ES node when `client.transport.sniff` was always being set to true. However this option works with against the 3 node cluster setup by audit-service-deploy.